File: server/models/Document.js
const mongoose = require('mongoose');

const DocumentSchema = new mongoose.Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true
    },
    filename: {
      type: String,
      required: true
    },
    originalName: {
      type: String,
      required: true
    },
    mimeType: {
      type: String,
      required: true
    },
    size: {
      type: Number,
      required: true
    },
    extractedText: {
      type: String,
      default: ''
    },
    summary: {
      type: String,
      default: ''
    },
    tags: {
      type: [String],
      default: []
    },
    chunks: [{
      text: String,
      chunkIndex: Number,
      embedding: { type: [Number], select: false } // hide embeddings by default to save memory
    }],
    documentType: {
      type: String,
      enum: ['general', 'syllabus', 'notes', 'textbook', 'assignment', 'research'],
      default: 'general'
    },
    subject: String,
    quizzes: [{
      question: String,
      options: [String],
      correctAnswer: Number,
      explanation: String
    }]
  },
  { timestamps: true }
);

DocumentSchema.index({ userId: 1, createdAt: -1 });

module.exports = mongoose.model('Document', DocumentSchema);

File: flutter_client/lib/features/documents/models/document.dart
class Document {
  final String id;
  final String originalName;
  final String mimeType;
  final num size;
  final String summary;
  final List<String> tags;
  final String documentType;
  final DateTime createdAt;

  Document({
    required this.id,
    required this.originalName,
    required this.mimeType,
    required this.size,
    this.summary = '',
    this.tags = const [],
    this.documentType = 'general',
    required this.createdAt,
  });

  factory Document.fromJson(Map<String, dynamic> json) {
    return Document(
      id: json['_id'] as String,
      originalName: json['originalName'] as String,
      mimeType: json['mimeType'] as String,
      size: json['size'] as num? ?? 0,
      summary: json['summary'] as String? ?? '',
      tags: (json['tags'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? [],
      documentType: json['documentType'] as String? ?? 'general',
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt'] as String) : DateTime.now(),
    );
  }
}
